TinCanTests/ActivityTest.cs

```diff
@@ -40,12 +40,16 @@ public void TestActivityIdCase()
         }
 
         [Test]
-        [ExpectedException("System.UriFormatException")]
         public void TestActivityIdInvalidUri()
         {
-            var activity = new Activity();
-            string invalid = "foo";
-            activity.id = invalid;
+            Assert.Throws<System.UriFormatException>(
```

Comment:
Ironically "but since no new features have been added to it for some time" is contradicted by the recent (in 3.2 this calendar year) addition of `Assert.ThrowsAsync`. See https://github.com/nunit/nunit/issues/464 There are various discussions in there about `Assert.That` that all pretty much call it a workaround at least in the async case.